src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -431,6 +512,96 @@ export class IndexPattern implements IIndexPattern {
     delete this.runtimeFieldMap[name];
   }
 
+  /**
+   * Add a runtime object and its subFields to the fields list
+   * @param name - The runtime object name
+   * @param runtimeObject - The runtime object definition
+   */
+  addRuntimeObject(name: string, runtimeObject: RuntimeObjectWithSubFields): IndexPatternField[] {
+    this.removeRuntimeObject(name);
+
+    const { script, subFields } = runtimeObject;
+
+    const fieldsCreated: IndexPatternField[] = [];
+
+    for (const [subFieldName, subField] of Object.entries(subFields)) {
+      const field = this.addRuntimeField(`${name}.${subFieldName}`, { ...subField, parent: name });
+      fieldsCreated.push(field);
+    }
+
+    this.runtimeObjectMap[name] = {
+      name,
+      script,
+      subFields: Object.keys(subFields),
+    };
+
+    return fieldsCreated;
+  }
+
+  /**
+   * Returns runtime object if exists
+   * @param name
+   */
+  getRuntimeObject(name: string): RuntimeObject | null {
+    return this.runtimeObjectMap[name] ?? null;
+  }
+
+  /**
+   * Returns runtime object (if exists) with its subFields
+   * @param name
+   */
+  getRuntimeObjectWithSubFields(name: string): RuntimeObjectWithSubFields | null {
```

Comment:
This method is not consumed in this PR but will be used when editing a runtime object. We'll get the runtime object + all its subFields (and their possible custom label and format) so the field editor can initiate its form.